test/src/test/resources/fixtures/hydration/hydration-with-interfaces-asking-typename.yml

```diff
@@ -0,0 +1,118 @@
+name: hydration with interfaces asking typename
+enabled:
+  current: false
+  nextgen: true
+overallSchema:
+  Issues: |
+    type Query {
+      nodes: [Node]
+    }
+    type JiraIssue implements Node @renamed(from: "Issue") {
+      id: ID! 
+    }
+    interface Node {
+      id: ID! 
+    }
+    type User implements Node {
+      id: ID! @hydrated(service: "Issues" field: "ariById" arguments: [{name: "id" value: "$source.id"}])
```

Comment:
I added a bunch of tests that ask for fields from interfaces where some impls of the interface have synthetic definitions i.e. hydrations, renames, deep renames.

This is in line with what Townsquare does with their schema i.e. `id: ID @renamed(from: "ari")` in some places where `id` is specified by `Node`.